src/modules/displaymanager/main.py

```diff
@@ -487,31 +487,45 @@ def set_autologin(self, username, do_autologin, default_desktop_environment):
             self.root_mount_point, "etc/lightdm/lightdm.conf"
             )
         text = []
+        addseat = False
+        loopcount = 0
 
         if os.path.exists(lightdm_conf_path):
             with open(lightdm_conf_path, 'r') as lightdm_conf:
                 text = lightdm_conf.readlines()
+                # Check to make sure [SeatDefaults] or [Seat:*] is in the config,
+                # otherwise we'll risk malforming the config
+                if not '[SeatDefaults]' in text and not '[Seat:*]' in text:
+                    addseat = True
 
             with open(lightdm_conf_path, 'w') as lightdm_conf:
+                if addseat:
+                    # Append Seat line to start of file rather than leaving it without one
+                    # This keeps the config from being malformed for LightDM
+                    text = ["[Seat:*]\n"] + text
+                loopcount = 0
                 for line in text:
                     if 'autologin-user=' in line:
                         if do_autologin:
                             line = "autologin-user={!s}\n".format(username)
                         else:
                             line = "#autologin-user=\n"
+                        text[loopcount] = line
+                    loopcount += 1
```

Comment:
Not sure why you're replacing the loop which writes line-by-line by one that updates the *text* array, then writes it in one go? That seems like a superfluous change.

Reply:
Originally I thought the loop was overwriting the file each time with the current line it was looped to, hence I came up with that loop, but now the established loop technique seems to be working for me as it was designed to so I restored it in the latest commit.